No more wallbangs/combat uppies

Onutsio opened the pull request:
## About The Pull Request

- You can no longer stun someone by throwing them into a wall unless they're already off-balance.
- You can no longer throw someone up(who's in cmode) if they're not laying down

Basically just adds back in some stuff from https://github.com/GeneralPantsuIsBadAtCoding/Azure-Peak/pull/344 that I suspect was removed during debloat on mistake.

## Testing Evidence

https://github.com/user-attachments/assets/f033bad2-1f6a-475e-8aea-1a15e8387e3d

https://github.com/user-attachments/assets/5013cc2a-8b79-4ce4-9425-0a362fb668c5


https://github.com/user-attachments/assets/8bfe62fa-a07e-4a90-94da-ba5b524dc6a8

## Why It's Good For The Game

I tire greatly of seeing people infinitely chain-stunned on a wall or instantly having their legs broken.

Permastuns are insanely bad and were already removed once:

https://github.com/user-attachments/assets/7b2f891c-8a0a-4b23-b8e4-dd0d4acc20b2

Non-giants can also do this, leaving a whopping 0.3 seconds of being in an aggro-grab and on the floor to do something(you will be stunned again)

https://github.com/user-attachments/assets/b319bfdc-21ca-4218-8229-b286692f2b51

Hemlock32 commented:
Can you still throw someone up a Z-level without them laying down? In non-combat situations, to help someone with low climbing? Just curious more than thinking its necessary to do

Onutsio commented:
> Can you still throw someone up a Z-level without them laying down? In non-combat situations, to help someone with low climbing? Just curious more than thinking its necessary to be able to do

No but they can just lay down for u! Also you can hold someone's hand and climb for them

FreeStylaLT commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -219,6 +220,8 @@
 								return
 						var/turf/start_T = get_turf(loc) //Get the start and target tile for the descriptors
 						var/turf/end_T = get_turf(target)
+						if(start_T.z != end_T.z && throwable_mob.mobility_flags & MOBILITY_STAND)
```
I'd suggest adding an istransparent(end_T) (and potentially a dist check?) check here instead, it'd allow you to still throw standing-up people over up, but still prevent _"combat uppies"_ by not allowing you to toss them into open air.

Onutsio replied:
You can just have them sit or grab their hand and climb. Not really neccesary imo

FreeStylaLT replied:
well you sorta made it !WORSE!

Mojamble commented:
Strongly against this, as these are some of the best counters to speed classes - not to mention they give unarmed builds some much needed "oomf" for those with the ability to pull it off.

Onutsio commented:
> Strongly against this, as these are some of the best counters to speed classes - not to mention they give unarmed builds some much needed "oomf" for those with the ability to pull it off.

They counter *every* class. As you can literally stunlock anyone

Onutsio commented:
Wallthrows require a passive grab and hitting r with giant. They arent very hard to land. Even normally they can stun cycle against a wall

burdwatcher commented:
Another way another server did it was making so it stunned if you weren't in combat mode, but you were otherwise fine if you were 'prepared' in cmode.

Might be a good trade off.

ProfanityTM commented:
> Wallthrows require a passive grab and hitting r with giant. They arent very hard to land. Even normally they can stun cycle against a wall
> 
> Every other server has already removed these. We had them removed before debloat.

I agree if they were removed previously, for as silly as I think it is, but. The whole giant throw? Legitimately, it happens so little that I've never even seen it. I had *no* idea, despite regularly throwing myself into PVP scenarios.

Mojamble commented:
I just find that getting into close range with a class that can juggle you is a risk you accept with PVP. I just dislike anything that limits options for the already underplayed monks and unarmed classes.

Onutsio commented:
> I just find that getting into close range with a class that can juggle you is a risk you accept with PVP. I just dislike anything that limits options for the already underplayed monks and unarmed classes.

Unarmed classes have plenty of options that arent infinite stunlocks against a wall. They survived before debloat, and will again.

Mojamble commented:
> t

Did they? Because I see them so rarely already. This'll be what kills it. Strongly oppose still, and this is as someone who suffers against this.

ProfanityTM commented:
> Unarmed classes have plenty of options that arent infinite stunlocks against a wall. They survived before debloat, and will again.

What exactly makes this different from being kicked into a wall repeatedly, other than the fact that the latter actually lets you use a two-handed weapon, and doesn't require you to switch intents or succeed on grabs/counter-grabs?

Onutsio commented:
> > Unarmed classes have plenty of options that arent infinite stunlocks against a wall. They survived before debloat, and will again.
> 
> What exactly makes this different from being kicked into a wall repeatedly, other than the fact that the latter actually lets you use a two-handed weapon, and doesn't require you to switch intents or succeed on grabs/counter-grabs?

You cannot stunlock someone against a wall with kicks because it doesnt actually inflict stun, just knock down. Further kicks do not inflict knockdown if you're already grounded. As demonstrated here. The victim can just get up and either parry or dodge the kick, or just move from the wall.

https://github.com/user-attachments/assets/332e4b98-6503-427d-aa12-035ee2708aca


Compared to literally perma stuncycling someone:

https://github.com/user-attachments/assets/6589fe43-71dd-4940-8d1a-36a2be596f42

ProfanityTM commented:
> You cannot stunlock someone against a wall with kicks because it doesnt actually inflict stun, just knock down. Further kicks do not inflict knockdown if you're already grounded. As demonstrated here. The victim can just get up and either parry or dodge the kick, or just move from the wall.

And why does throwing someone into a wall stun them, but not kicking? That sounds like the actual problem here, if anything. It shouldn't, from the get-go. I'd argue it still serves it's usefulness in downing an opponent as an alternative to kicking, and shouldn't be removed, but changed.

Onutsio commented:
> If anything, giants SHOULDN'T be able to throw you with a passive grab either, considering how powerful throws are either way.

Non-giants can do this too, just slightly less effectively(leaves the victim a 0.3 second window rather than literally nothing, mind you, every throw drops all your weapons and you're in an aggro grab during that 0.3 seconds)

https://github.com/user-attachments/assets/4a8f9dac-ec37-4f1b-99b8-65b9e09318ff

ProfanityTM commented:
> > If anything, giants SHOULDN'T be able to throw you with a passive grab either, considering how powerful throws are either way.
> 
> Non-giants can do this too, just slightly less effectively(leaves the victim a 0.3 second window rather than literally nothing, mind you, every throw drops all your weapons and you're in an aggro grab during that 0.3 seconds)
> 
>  AZURE.PEAK.2025-04-06.02-00-15.mp4

I'm aware of how regular people can throw, and that's not the issue. The stupid part for giants is the passive grab throw. It's much easier to passively throw someone, than to reinforce the grab because of struggle/resist mechanics.

Dakken243 commented:
Removing infinite stunlock is a godsend.

Onutsio commented:
> Can you still throw someone up a Z-level without them laying down? In non-combat situations, to help someone with low climbing? Just curious more than thinking its necessary to be able to do

Changed mind, implemented. Cmode is what turns off standing uppies

Lutowski commented:
yeah I think I nuked this on accident, good catch 👍

Mojamble commented:
It's so over..
